### ISSUE TITLE
Suggest closing VMs when "not enough memory to start VM"

### ISSUE THREAD

**Issue by deeplow** [QubesOS/qubes-issues#7327]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

When a user opens too many VMs, they get the following error message
![failed-to-start](https://user-images.githubusercontent.com/47065258/157040333-1a0c6cd9-96d3-4686-90b7-d6ebb8c0c56f.png)

This error makes people think there is something wrong with the system, when this is simply the way Qubes.

### The solution you'd like

Following this design heuristic:
> **Help users recognize, diagnose, and recover from errors**
> Error messages should be expressed in plain language (no codes), precisely indicate the problem, and constructively suggest a solution.

I would propose that when a user tries to start a new qube, they be presented with a more informative message:

> There is not enough memory to start a new qube. Which qubes would you like to **shut down**?
> - personal
> - work
> - vault
> - disp6161

This would present the solution instead of just an error.

### The value to a user, and who that user might be

This is something all users will face very early on in their Qubes usage and non-technical users will probably have a hard time understanding what they needs to do. As I summarized in my talk on the Qubes summit:

![](https://user-images.githubusercontent.com/47065258/157040686-7a0233a1-3e16-4c30-aaf5-0e891d8998f6.png)

Also, this would be a great opportunity to train user's mental models. When they have too many qubes open, they are taught to shut them down instead of just being left on their own.

**Comment by andrewdavidwong** [comment]
I agree with your general idea and reasoning, but I don't necessarily agree that shutting down a running qube is always the right solution. In some cases, the user's informed reaction will be, "Ohh, I get it now. Well, in that case, never mind. I just want to keep using the qubes I already have running for a bit longer until I finish my current task."

@ninavizz, just letting you know that this is a UX issue.

**Comment by deeplow** [comment]
Yep. The ability to say no is important.

This could even be something as simple a message warning the user about the issue and what the solution might be, with a "don't show show this again". Even that would probably help a lot.

**Comment by marmarek** [comment]
Another option is just closing some applications but keep their qubes running - that's often enough.

**Comment by ben-grande** [comment]
For contributors, search for "Not enough memory" in qubes-core-admin.
